### ISSUE TITLE
destination incorrectly set to stdout when source is set

### ISSUE THREAD

**Issue by mdame** [perltidy/perltidy#115]
Prior to version 20221111, setting source to a file would correctly set the output to a file named sourcefilename.tdy. Beginning with version 20221111, the output goes to stdout instead:

```
  # Output file will be test.pl.tdy prior to version 20221111. Now the output goes to stdout
  Perl::Tidy::perltidy(source => "test.pl");

```
The problem seems to be that the call to process_all_files() sends %input_hash and process_all_files() sets its own version of $source_stream to $rinput_hash->{'source'}. The code that runs before calling process_all_files() that sets @Arg_files to $source_stream correctly sets $source_stream to undef, but process_all_files() uses its own version.

I'm not familiar enough with the internals to be comfortable providing a full patch, but setting $input_hash{'source'} = undef seems to be sufficient:

```
    # Tidy.pm line 923:
    # no filenames should be given if input is from an array
    if ($source_stream) {
        if ( @Arg_files > 0 ) {
            Die(
"You may not specify any filenames when a source array is given\n"
            );
        }

        # we'll stuff the source array into Arg_files
        unshift( @Arg_files, $source_stream );

        # No special treatment for source stream which is a filename.
        # This will enable checks for binary files and other bad stuff.
        $source_stream = $input_hash{'source'} = undef unless ref($source_stream);
    }
```

**Comment by shancock9** [comment]
Thanks for the report and diagnosis. Your fix is correct. This was an oversight when the code was rewritten as part of an update to clean up the coding for character encoding. The updated value of $source_stream was not getting passed down the line.  I made the patch slightly differently in order to preserve the input_hash for possible diagnostic use.

This seems to be a little-used method for calling the module, which may be why the problem hasn't been reported sooner. (Usually both source and destination are specified). If this is causing a problem, I can do a release to CPAN in the near future. Let me know.

**Comment by mdame** [comment]
I'm in no rush for an official fix. I'm running with my patched version at the moment, which solves my immediate needs. Thanks!

**Comment by shancock9** [comment]
This is fixed in version 20230701 at CPAN
